pinecone/config/config.py

```diff
@@ -46,10 +46,10 @@ def build(
         if not host:
             raise PineconeConfigurationError("You haven't specified a host.")
 
-        openapi_config = (
-            openapi_config
-            or kwargs.pop("openapi_config", None)
-            or OpenApiConfigFactory.build(api_key=api_key, host=host)
-        )
+        if openapi_config:
+            openapi_config.host = host
+            openapi_config.api_key = {"ApiKeyAuth": api_key}
```

Comment:
When the user provides this object with some configuration in it, we want to merge the api key and host settings into it rather than building a fresh object.